src/IIIFPresentation/DLCS/Exceptions/DlcsException.cs

```diff
@@ -1,15 +1,22 @@
+using System.Net;
+using Core;
+
 namespace DLCS.Exceptions;
 
 /// <summary>
 /// Represents an error interacting with DLCS
 /// </summary>
 public class DlcsException : Exception
 {
-    public DlcsException(string? message) : base(message)
+    public HttpStatusCode? StatusCode { get; }
```

Comment:
Can this + the 2 ctor params be non-nullable? Looks like it's always set and would save null check/coalescing when verifying response